stacker_blueprints/eks.py

```diff
@@ -0,0 +1,427 @@
+from awacs.aws import (
+    Allow,
+    Statement,
+    Principal,
+    Policy
+)
+from awacs.sts import (
+    AssumeRole
+)
+
+from troposphere import (
+    Base64,
+    FindInMap,
+    GetAtt,
+    Join,
+    Ref,
+    Output
+)
+
+from troposphere.autoscaling import (
+    AutoScalingGroup,
+    LaunchConfiguration,
+    Tag
+)
+
+from troposphere.ec2 import (
+    BlockDeviceMapping,
+    EBSBlockDevice
+)
+
+from troposphere.iam import (
+    InstanceProfile,
+    Role
+)
+
+from troposphere.policies import (
+    AutoScalingReplacingUpdate,
+    AutoScalingRollingUpdate,
+    UpdatePolicy
+)
+
+from troposphere import eks
+from stacker.blueprints.base import Blueprint
+
+
+class Cluster(Blueprint):
+    VARIABLES = {
+        "Name": {
+            "type": str,
+            "description": "The name of the cluster to create.",
+        },
+        "ExistingRoleArn": {
+            "type": str,
+            "description": "IAM Role ARN with EKS assume role policies. One "
+                           "will be created if it's not provided.",
+            "default": ""
+        },
+        "Version": {
+            "type": str,
+            "description": "Kubernetes version",
+            "default": "",
+        },
+        "SecurityGroupIds": {
+            "type": str,
+            "description": "A comma separated list of security group ids for "
+                           "controlling ENI access from EKS to workers",
+        },
+        "SubnetIds": {
+            "type": str,
+            "description": "A comma separated list of subnet ids where you "
+                           "will launch your worker nodes",
+        }
+    }
+
+    # This creates an IAM role which EKS requires, as described here:
+    # https://docs.aws.amazon.com/eks/latest/userguide/getting-started.html#eks-create-cluster
+    def create_iam_role(self):
+        eks_service_role_id = "EksServiceRole"
+        t = self.template
+        t.add_resource(
+            Role(
+                eks_service_role_id,
+                AssumeRolePolicyDocument=Policy(
+                    Statement=[
+                        Statement(
+                            Effect=Allow,
+                            Action=[AssumeRole],
+                            Principal=Principal("Service",
+                                                ["eks.amazonaws.com"])
+                        )
+                    ]
+                ),
+                Path="/",
+                ManagedPolicyArns=[
+                    "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy",
+                    "arn:aws:iam::aws:policy/AmazonEKSServicePolicy",
+                ]
+            )
+        )
+        return GetAtt(eks_service_role_id, "Arn")
```

Comment:
You also probably want to include some outputs for the created role.  It may not be used now, but always a good idea.  These days I tend to create outputs for just about every `Ref` and `GetAtt` that a resource provides, just to avoid any issues in the future.

Reply:
It is output later - whether or not existing role is passed in, or a new one is created, it's an output in create_stack.